Login page #1

Oress opened the pull request.

Oress committed:
added login form, signup form, login page

Oress committed:
deleted some lines

horba commented on Gruvo/Gruvo/ClientApp/src/app/login-form/login-form.component.html:
```diff
@@ -0,0 +1,11 @@
+    <div class="loginform  form-inline">
```
it's better class="login-form  form-inline"

horba commented on Gruvo/Gruvo/ClientApp/src/app/login-form/login-form.component.ts:
```diff
@@ -0,0 +1,17 @@
+
+  constructor() {
+  }
+  ngOnInit() {
```
please, don't use empty hooks. add it if nesessary

horba commented on Gruvo/Gruvo/ClientApp/src/app/login-form/login-form.component.ts:
```diff
@@ -0,0 +1,17 @@
+export class LoginFormComponent implements OnInit {
+
+  email: string;
+  pass: string;
```
please, don't use so shortcuts

horba commented on Gruvo/Gruvo/ClientApp/src/app/login-page/login-page.component.css:
```diff
@@ -0,0 +1,48 @@
+  display: flex;
+  flex-flow: column;
+  align-items: stretch;
+  height: 100vh;
```
why height? are you sure?

horba commented on Gruvo/Gruvo/ClientApp/src/app/login-page/login-page.component.css:
```diff
@@ -0,0 +1,48 @@
+  background-color: rgb(51, 51, 102);
+  height: 80px;
+  width: 100%;
+  padding-top: 1em;
```
why do you use em?

horba commented on Gruvo/Gruvo/ClientApp/src/app/login-page/login-page.component.css:
```diff
@@ -0,0 +1,48 @@
+  justify-content:center;
+  flex-direction: row;
+  padding-bottom: 20%;
+  background: linear-gradient( rgba(35,43,85,0.75), rgba(35,43,85,0.95)), url(https://dl.dropboxusercontent.com/u/22006283/preview/codepen/clouds-cloudy-forest-mountain.jpg) no-repeat center center;
```
think about other ways for image's usage

horba commented on Gruvo/Gruvo/ClientApp/src/app/login-page/login-page.component.ts:
```diff
@@ -0,0 +1,13 @@
+export class LoginPageComponent implements OnInit {
+  constructor() {
+  }
+  ngOnInit() {
```
empty hook

horba commented on Gruvo/Gruvo/ClientApp/src/app/signup-form/signup-form.component.html:
```diff
@@ -0,0 +1,24 @@
+    <input id="Password" [(ngModel)]="pass" type="password" class="form-control" placeholder="Password" maxlength="50">
+  </div>
+
+  <!--<div class="form-group">
```
remove unusable comments or uncomment it if it is nessesary

horba commented on Gruvo/Gruvo/ClientApp/src/app/signup-form/signup-form.component.ts:
```diff
@@ -0,0 +1,18 @@
+
+  login: string;
+  email: string;
+  pass: string;
```
naming

horba commented on Gruvo/Gruvo/ClientApp/src/app/signup-form/signup-form.component.ts:
```diff
@@ -0,0 +1,18 @@
+  email: string;
+  pass: string;
+
+  constructor() {
```
empty unusable constructor

horba commented on Gruvo/Gruvo/ClientApp/src/app/signup-form/signup-form.component.ts:
```diff
@@ -0,0 +1,18 @@
+
+  constructor() {
+  }
+  ngOnInit() {
```
empty

horba commented on Gruvo/Gruvo/ClientApp/src/app/signup-form/signup-form.component.html:
```diff
@@ -0,0 +1,24 @@
+  <h1>Sign Up</h1>
+  <div class="form-group">
+    <label for="Login">Login:</label>
+    <input id="Login" [(ngModel)]="login" type="text" class="form-control" placeholder="Login" maxlength="50">
```
is this validation of max length corresponds with validation on server/db or other reasons?

horba commented on Gruvo/Gruvo/ClientApp/src/app/login-page/login-page.component.html:
```diff
@@ -0,0 +1,26 @@
+        </p>
+
+      </div>
+      <gr-signup-form class="container"></gr-signup-form>
```
it's not good idea set some class for component by this way. In this case it's simpler and better input gr-signup-form into div or other wrapper and set class="container" for this wrapper

as I understood, it means that login and registration are located on the same rule?
It seems like we should have login page with **link** to registration which reference to other page where the sign up is

horba commented on Gruvo/Gruvo/ClientApp/src/app/login-page/login-page.component.html:
```diff
@@ -0,0 +1,26 @@
+<link href="https://fonts.googleapis.com/css?family=Kaushan+Script" rel="stylesheet">
```
please use angular-cli

Oress committed:
update 1

moved font import link to style.css, removed empty constructors and ngOninit methods, removed validation parts from signup component

Oress committed:
update 2

Depart header and footer from login/signup component

Oress committed:
added pictures

Oress committed:
edited copyright

Oress committed:
added readme, removed footer and header components for now

Oress committed:
Revert "added readme, removed footer and header components for now"

This reverts commit 3e638bf400ac87c1e8cee44bd5987f573480c1a6.

Oress committed:
update 4

Added header and footer back clipped them to top and bottom of the screen, changed height to min-height

Oress committed:
main layout

Co-Authored-By: Mariia Dehtiarenko <mdegtyarenko725@gmail.com>

Oress committed:
deleted imports

Co-Authored-By: Mariia Dehtiarenko <mdegtyarenko725@gmail.com>

Oress committed:
Revert "deleted imports"

This reverts commit f71d527182d80061d0426911a65f82675f0eac07.

likash committed:
conflicts were resoleved

likash committed:
resolved conflicts

Oress committed:
update #5

added depencies

Oress committed:
removed screenshots

Oress committed:
added login service
